pkg/istio-agent/xds_proxy.go

```diff
@@ -367,29 +397,73 @@ func (p *XdsProxy) handleUpstreamResponse(con *ProxyConnection) {
 					TypeUrl:       v3.NameTableType,
 					ResponseNonce: resp.Nonce,
 				}
+			case v3.ExtensionConfigurationType:
+				// intercepts ECDS, replace Wasm remote load with local file path.
+				if dr.wasmExtensionConverted {
```

Comment:
The split paths here is a bit confusing on first read - makes sense after digging into it though. One question - is the behavior changed if we call forwardToEnvoy() directly in handleUpstreamECDSResponse instead of sending back on responsesChan? That may simplify this a bit?

Reply:
I wrote it that way initially, but find it might run into some corner case that will block the channel. So if the conversion goroutine finishes, but the connection has already been teared down, Send to envoy will result in error, which I think could potentially block on the [error channel](https://github.com/istio/istio/blob/770c35cd8bb793a375263d55259e935437e846f6/pkg/istio-agent/xds_proxy.go#L377) since reader should have already stopped, and also will generate a log indicates downstream sends error, which is confusing? Writing it in this way will cause the update simply being dropped, which is fine since the next full push on reconnect will have this update anyway. Not sure if I am overthinking on this.

Reply:
I am not too against this, but I suspect we could workaround that if you prefer. For example, we can check if the error channel is blocking with `select{ default: }` or see if the connection is closed (not sure how, but I am sure we have this info somewhere) before sending?

I am fine either way though

Reply:
I updated to unblock on the error channel. Thanks!